### PR TITLE
Feat(eos_designs): Allow disabling filtering on redist connected in underlay bgp

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Allow disabling filtering on redist connected in underlay bgp

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

```yaml
# Filter redistribution of connected into underlay routing protocol.
# Only applicable when overlay_routing_protocol != 'none' and underlay_routing_protocol == BGP.
# Creates a route-map and prefix-list assigned to redistribute connected permitting only loopbacks and inband management subnets.
underlay_filter_redistribute_connected: < true | false | default -> true >
```

## How to test

Added to molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/route_maps.py [hunk 1]
```diff
@@ -119,6 +109,22 @@ def _route_maps_vrf_default(self) -> dict | None:
                 "match": ["ip address prefix-list PL-STATIC-VRF-DEFAULT"],
             }
 
+        if self._underlay_filter_redistribute_connected:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/route_maps.py]: Need to rebase to take into account the new route-map syntax - maybe let's keep the ordering of generation (RM-CONN before RM-EVPN if both present?)

**Review by gmuloc — COMMENTED** [review]
LGTM overall - just need following the latest changes in devel

**Review by gmuloc — APPROVED** [review]
LGTM

**Review by emilarista — APPROVED** [review]
LGTM!

**Pull request merged**